src/test/java/com/datalogics/pdf/samples/extraction/TextExtractTest.java

```diff
@@ -40,4 +64,46 @@ public void testExtractTextReadingOrder() throws Exception {
         final String extractedText = contentsOfTextFile(file);
         assertEquals(contentsOfResource(EXTRACTED_DOCUMENT_NAME), extractedText);
     }
+
+    @Test
+    public void testDocumentWithNoText() throws Exception {
+        // Create a new document with a single empty page
+        final PDFDocument document = createEmptyPdf();
+        final File emptyPdf = newOutputFileWithDelete(EMPTY_PDF_FILE_PATH);
+        final RandomAccessFile outputPdfFile = new RandomAccessFile(emptyPdf, "rw");
+        final ByteWriter outputWriter = new RandomAccessFileByteWriter(outputPdfFile);
+        document.saveAndClose(outputWriter, PDFSaveFullOptions.newInstance());
+
+        final File emptyTxt = newOutputFileWithDelete(EMPTY_TEXT_FILE_PATH);
+
+        final URL inputUrl = emptyPdf.toURI().toURL();
+        final URL outputUrl = emptyTxt.toURI().toURL();
+
+        final ArrayList<LogRecord> logRecords = new ArrayList<LogRecord>();
+        final Logger logger = Logger.getLogger(TextExtract.class.getName());
+        try (LogRecordListCollector collector = new LogRecordListCollector(logger, logRecords)) {
+            TextExtract.extractTextReadingOrder(inputUrl, outputUrl);
+        }
+
+        // Verify that we got the expected log message
+        assertEquals("Must have one log record", 1, logRecords.size());
+        final LogRecord logRecord = logRecords.get(0);
+        assertEquals(inputUrl.getFile() + " did not have any text to extract.", logRecord.getMessage());
```

Comment:
To avoid getting %-escapes in the filename, it's better to use ``inputUrl.toURI().getPath()`` here.